liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/PluginConfigXmlDocument.java

```diff
@@ -0,0 +1,191 @@
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.maven.model.Profile;
+import org.codehaus.mojo.pluginsupport.util.ArtifactItem;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class PluginConfigXmlDocument {
+    
+    private Document doc;
+    
+    private PluginConfigXmlDocument() {    
+    }
+    
+    public static PluginConfigXmlDocument newInstance(String rootElement) throws ParserConfigurationException {
+        PluginConfigXmlDocument configDocument = new PluginConfigXmlDocument();
+        configDocument.createDocument(rootElement);
+        return configDocument;
+    }
+    
+    public void createDocument(String rootElement) throws ParserConfigurationException {
+        DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+        doc = docBuilder.newDocument();
+        doc.setXmlStandalone(true);
+        Element element = doc.createElement(rootElement);
+        doc.appendChild(element);
+    }
+    
+    public void createElement(String key, boolean value) {
+        createElement(doc.getDocumentElement(), key, Boolean.toString(value));
+    }
+    
+    public void createElement(Element element, String key, boolean value) {
+        createElement(element, key, Boolean.toString(value));
+    }
+    
+    public void createElement(String key, File value) throws IOException {
+        if (value == null) {
+            return;
+        } else {
+            createElement(doc.getDocumentElement(), key, value.getCanonicalPath());
+        }
+    }
+    
+    public void createElement(String name, Map<String, String> values) {
+        if (values == null || values.isEmpty()) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        for (Map.Entry<String, String> entry : values.entrySet()) {
+            createElement(child, entry.getKey(), entry.getValue());
+        }
+        doc.getDocumentElement().appendChild(child);
+    }
+    
+    public void createElement(String name, List<String> values) {
+        if (values == null || values.isEmpty()) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        for (int i = 0; i < values.size(); i++) {
+            createElement(child, "param", values.get(i));
+        }
+        doc.getDocumentElement().appendChild(child);
+    }
+    
+    public void createElement(String name, ArtifactItem value) {
+        if (value == null) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        createElement(child, "groupId", value.getGroupId());
+        createElement(child, "artifactId", value.getArtifactId());
+        createElement(child, "version", value.getVersion());
```

Comment:
Maybe it should take the same approach as Install object unless some default version is available.